packages/mocha-debug-reporter/src/index.ts

```diff
@@ -0,0 +1,203 @@
+/**
+ * External dependencies
+ */
+import Mocha from 'mocha';
+import fs from 'fs';
+import path from 'path';
+
+import type { Runner, MochaOptions, Suite, Hook, Test } from 'mocha';
+
+const {
+	EVENT_RUN_BEGIN,
+	EVENT_RUN_END,
+	EVENT_DELAY_BEGIN,
+	EVENT_DELAY_END,
+
+	EVENT_SUITE_BEGIN,
+	EVENT_SUITE_END,
+	EVENT_HOOK_BEGIN,
+	EVENT_HOOK_END,
+	EVENT_TEST_BEGIN,
+	EVENT_TEST_END,
+	EVENT_TEST_FAIL,
+	EVENT_TEST_PASS,
+} = Mocha.Runner.constants;
+
+interface Event {
+	timestamp: Date;
+	name: string;
+	data?: Record< string, unknown >;
+}
+
+export = class DebugReporter extends Mocha.reporters.Base {
```

Comment:
`export =` is used to generate a CJS module (https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require)